File: 백준/Bronze/32046. Snacks within 300 Yen/Snacks within 300 Yen.py
import sys
input=lambda:sys.stdin.readline().rstrip()
while True:
     n=int(input())
     ans=0
     if n==0: break
     arr=[*map(int,input().split())]
     for i in range(n):
          if ans+arr[i]<301:
               ans+=arr[i]
     print(ans)

File: 백준/Bronze/32046. Snacks within 300 Yen/Snacks within 300 Yen.cc
#include <iostream>
using namespace std;
int main(void) {
    int n, ans, x;
    while (true) {
        cin >> n;
        ans = 0;
        if (n == 0) break;
        while (n--) {
            cin >> x;
            if (ans + x < 301)
                ans += x;
        }
        cout << ans << "\n";
    }
    return 0;
}